discussment-core/src/main/java/org/danekja/discussment/core/service/imp/DefaultPostReputationService.java

```diff
@@ -21,53 +18,73 @@
  */
 public class DefaultPostReputationService implements PostReputationService {
 
-    private PostReputationDao postReputationDao;
     private UserPostReputationDao userPostReputationDao;
+    private PostDao postDao;
 
     private AccessControlService accessControlService;
     private DiscussionUserService userService;
 
-    public DefaultPostReputationService(PostReputationDao postReputationDao,
-                                        UserPostReputationDao userPostReputationDao,
+    public DefaultPostReputationService(UserPostReputationDao userPostReputationDao,
+                                        PostDao postDao,
                                         DiscussionUserService userService,
                                         AccessControlService accessControlService){
 
-        this.postReputationDao = postReputationDao;
         this.userPostReputationDao = userPostReputationDao;
+        this.postDao = postDao;
 
         this.userService = userService;
         this.accessControlService = accessControlService;
     }
 
-    public PostReputation createPostReputation (PostReputation entity){
-        return postReputationDao.save(entity);
+    public synchronized void addLike(Post post){
+        // todo: only one service must exist in the app
```

Comment:
Solution to the synchronization issue and multiple PostReputationService instances:

Create global static variable of type Object and use its monitor to lock:
```
private static final Object monitor = new Object();
```

then use it in methods like this:
```
public void addLike(Post post) {
    synchronized(monitor) {
         //your code here
    }
}
```